src/lib.rs

```diff
@@ -21,13 +21,31 @@ impl rlp::Encodable for Node {
     fn rlp_append(&self, s: &mut rlp::RlpStream) {
         match self {
             Node::Leaf(ref k, ref v) => {
-                let l: Vec<u8> = k.clone().into();
+                let key_prefixed = add_indicator_prefix(k.clone().into(), true);
```

Comment:
@cdetrio could you please check this? I think there is an error here, because you are calling `into` which because of the signature of `add_indicator_prefix` will call the conversion function from `NibbleKey` to `Vec<u8>` which makes it a byte vector. I can fix the thing, I just need you to confirm that there is indeed an issue here.